apis/bigqueryconnection/v1beta1/connection_reference.go

```diff
@@ -149,8 +150,32 @@ func (r *BigQueryConnectionConnectionRef) NormalizedExternal(ctx context.Context
 		return "", fmt.Errorf("reading status.externalRef: %w", err)
 	}
 	if actualExternalRef == "" {
-		return "", fmt.Errorf("BigQueryConnectionConnection is not ready yet.")
+		return "", fmt.Errorf("BigQueryConnectionConnection is not ready yet")
 	}
 	r.External = actualExternalRef
 	return r.External, nil
 }
+
+func ResolveCloudSQLReference(ctx context.Context, reader client.Reader, namespace string, ref *iamv1beta1.MemberReference) (string, error) {
+	key := types.NamespacedName{Name: ref.Name, Namespace: ref.Namespace}
+	if key.Namespace == "" { // use the namespace of the IAM resource if a namespace is not provided in the reference
+		key.Namespace = namespace
+	}
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(BigQueryConnectionConnectionGVK)
+	if err := reader.Get(ctx, key, u); err != nil {
+		if apierrors.IsNotFound(err) {
+			return "", k8s.NewReferenceNotFoundError(u.GroupVersionKind(), key)
+		}
+		return "", fmt.Errorf("reading referenced %s %s: %w", BigQueryConnectionConnectionGVK, key, err)
+
+	}
+	sa, found, err := unstructured.NestedString(u.Object, "status", "observedState", "cloudSQL", "serviceAccountID") // TODO: a better way to provide info on which field to read?
+	if err != nil {
+		return "", err
```

Comment:
Would GCP service return empty SAID? If so, this would be found=true, err=nil, and sa=""

Reply:
During my testing, I did not encounter an empty service account. Based on the golden object files generated from our existing tests, service accounts are always present for connections of type Spark, CloudSQL, and CloudResource. Did you encounter some corner case?